src/Exttload.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import {IExttload} from "./interfaces/IExttload.sol";
+import {Extsload} from "./Extsload.sol";
+
+/// @notice Enables public transient storage access for efficient state retrieval by external contracts.
+/// https://eips.ethereum.org/EIPS/eip-2330#rationale
+abstract contract Exttload is IExttload, Extsload {
```

Comment:
hmm i think Exttload/IExttload should be decoupled from Extsload

while they are functionally similar, one of the reasons why we moved Extsload to a different file (with MIT) was so other projects can use the same logic

by decoupling it, other projects can pick and choose if they need Exttload, Extsload, or both

Reply:
fair point that a project might only want to use Exttload, but not Extsload, should I also add `exttload(bytes32 startSlot, uint256 nSlots) external view returns (bytes memory)` even though we are not using it internally?

Reply:
since tstore is so new, im not sure if other projects will be using tstore in a sequential manner?

is it possible tstore a struct that exceeds a word / 256 bits 🤔 

i will say that sparse exttload would be interesting (i.e. read multiple reserve amounts in a single call)

i.e. the exttload equilvalent to https://github.com/Uniswap/v4-core/pull/625